vignettes/adlb.Rmd

```diff
@@ -0,0 +1,324 @@
+ $$ FLI = \left[\frac{\exp(\lambda)}{1 + \exp(\lambda)}\right] \times 100 $$
+where $$ \lambda = 0.953 \times \ln (\text{triglycerides}) + 0.139 \times BMI + 0.718 \times \ln (GGT) + 0.053 \times \text{waist circumference} – 15.745. $$
+FLI is interpreted as a score between 0 and 100, where higher values indicate a worse condition.
+
+*   **Homeostasis Model Assessment – Insulin Resistance (HOMA-IR)**: Calculated from fasting plasma glucose (mmol/L) and fasting plasma insulin (mIU/L).
+ $$ \text{HOMA-IR} = \frac{FPI \times FPG}{22.5}. $$
+An alternative formula is used when fasting plasma glucose is assessed in mg/dL: $$ \text{HOMA-IR} = \frac{FPI \times FPG}{405}. $$
+
+To derive these parameters and scores, the following variables will be needed:
+
+*   From ADLB: TRIG (Triglycerides), GGT, GLUC (fasted), INSULIN (fasted).
+*   From ADVS: BMI, WSTCIR (Waist Circumference).
+
+## Merge ADVS variables {#merge}
+In this section, we will merge relevant variables from the ADVS (Analysis Dataset for Vital Signs) dataset into the ADLB dataset. This is necessary to include variables such as BMI and Waist Circumference, which are required for the derivation of FLI score parameter in subsequent section.
+
+```{r}
+# Load ADVS dataset (assuming it has been created by ad_advs.R)
+admiralmetabolic_advs <- admiralmetabolic::admiralmetabolic_advs
+advs <- convert_blanks_to_na(admiralmetabolic_advs)
+
+# Merge BMI and WSTCIR from ADVS to ADLB based on subject and date
+adlb <- adlb %>%
+  derive_vars_transposed(
+    advs,
+    by_vars = exprs(!!!get_admiral_option("subject_keys"), ADT),
+    key_var = PARAMCD,
+    value_var = AVAL,
+    filter = PARAMCD %in% c("BMI", "WSTCIR")
+  )
+```
+
+Please note that in this example we merge on ADT, but it is also possible to merge on AVISIT or other windowing algorithms specified in the statistical analysis plan.
+
+```{r echo=FALSE}
+# Example display of derived data
+dataset_vignette(
+  arrange(adlb, USUBJID, ADY, PARAMN),
+  display_vars = exprs(!!!get_admiral_option("subject_keys"), PARAMCD, PARAM, AVAL, ADT, ADY, AVISIT, AVISITN, BMI, WSTCIR)
+)
+```
+
+## Derive Metabolic Parameters {#derive_metabolic}
+
+In this section, we will derive the metabolic parameters and scores described previously by directly implementing the calculation formulas.
+
+### Homeostasis Model Assessment – Insulin Resistance (HOMA-IR)
+
+This part describes how to calculate the HOMA-IR index from fasting plasma glucose and fasting plasma insulin. This derivation does not require merging data from other datasets.
+
+```{r}
+# Derive HOMA-IR using derive_param_computed
+adlb <- adlb %>%
+  derive_param_computed(
+    by_vars = exprs(!!!get_admiral_option("subject_keys"), AVISIT, AVISITN, ADT, ADY, !!!adsl_vars),
+    parameters = c("INSULIN", "GLUC"),
+    set_values_to = exprs(
+      AVAL = if_else(
+        complete.cases(AVAL.INSULIN, AVAL.GLUC),
```

Comment:
Will not `derive_vars_computed` automatically handle this? 

I've been doing some testing, and I think we can keep it as:

```r
 AVAL = AVAL.INSULIN * AVAL.GLUC / 22.5,
 
```